enterprise/coderd/prebuilds/reconcile.go

```diff
@@ -128,6 +140,17 @@ func (c *StoreReconciler) Stop(ctx context.Context, cause error) {
 		return
 	}
 
+	// Unregister the metrics collector.
+	if c.metrics != nil && c.registerer != nil {
+		if !c.registerer.Unregister(c.metrics) {
```

Comment:
If register failed during New, will this log the error here? (Seems like additional noise if yes.)

Reply:
Probably yes, but it's just one more log message so it's likely not worth the instrumentation to prevent it in my opinion. WDYT?